pkg/controller/controller_binding.go

```diff
@@ -266,13 +294,16 @@ func (c *controller) reconcileBinding(binding *v1alpha1.Binding) error {
 		if err != nil {
 			s := fmt.Sprintf("Error deleting secret: %s", err)
 			glog.Warning(s)
-			c.updateBindingCondition(
+			if err := c.updateBindingCondition(
 				binding,
 				v1alpha1.BindingConditionReady,
 				v1alpha1.ConditionUnknown,
 				errorEjectingBindReason,
 				errorEjectingBindMessage+s,
-			)
+			); err != nil {
+				glog.Errorf("updating binding condition (%s)", err)
```

Comment:
if we fail below, do we come back through here and fail to eject the binding? (as it is already ejected?) 

How does  ConditionUnknown play into this?